feat: bridge modal UI and network switch for mainnet and testnet

0xthi opened the pull request:
-Added both testnet and mainnet switch
-Bridge Modal completed for base sepolia
-Fixed code redundancy issues
-Networks can be added automatically if not vailable in metamask
-Deposit and Withdraw components are disabled for wrong networks

0xthi committed:
across bridge, get quote

0xthi committed:
Across bridge+action component

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into thileep/bac-123-across-integration-for-cross-chain-bridging

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into thileep/bac-123-across-integration-for-cross-chain-bridging

0xthi committed:
refactor: Replaced deposit with PermitDeposit in Across

PermitDeposit implemented instead of normal deposit in acrossBridge.

This change will enable permit2 signing in across cross chain intent

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into ui-225-actual-order-amount-is-not-editable

0xthi committed:
feat: Added network switch and bridge UI

Network switch added, components will be disabled for wrong network and bridge UI completed

It solves the issue that app only works with supported network and bridge networks

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into ui-153-add-conditional-rendering-logic-on-the-app-based-on-network

0xthi committed:
fix: Build error fix

refactored code to solve syntax and dependency error

Build error resolved

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into ui-153-add-conditional-rendering-logic-on-the-app-based-on-network

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into ui-153-add-conditional-rendering-logic-on-the-app-based-on-network

0xthi committed:
feat: added arbitrum mainnet and feature to add chain in metamask

arbitrum one mainnet is supported. Users can add networks directly to metamask if it is not available

It solves wallet switching issues and adds arbitrum mainnet

0xthi committed:
fix: reduced code redundancy issues

Move network states to single file and reduced code redundancy

0xthi committed:
fix: created wrapper modal for deposit button

it solves code redundancy

Prithvi101 commented on src/app/trade/page.tsx:
```diff
@@ -52,13 +55,7 @@ export default function TradePage() {
-                    >
-                        Deposit
-                    </Button>
+                    <DepositButtonWrapper variant="trade" />
```
you can remove it from here and add it to DailogWrapper component

Prithvi101 commented on src/components/modals/BridgeComponent.tsx:
```diff
@@ -1,158 +1,301 @@
-                  </pre>
+
+    return (
+        <div className='flex flex-col gap-6'>
```
You can use BaseModal Component and take reference from addfundsmodal

Prithvi101 commented on src/components/modals/DepositButtonWrapper.tsx:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    return (
+        <SignInRedirectorLayout>
```
you can use dailogue wrapper instead

0xthi replied:
I've made the above changes. Could you review it pls

0xthi committed:
fix: pr review changes fixed

Removed DepositWrapper and refactored and used DialogsWrapper. In BridgeComponent used BaseModal component

Prithvi101 commented on src/components/layouts/DialogsWrapper.tsx:
```diff
@@ -1,35 +1,53 @@
             {isAddFundsModalOpen && <AddFundsModal />}
 
-            {children}
+            {children || (
```
not understanding the purpose of this  why are we making some optional component with children

Prithvi101 commented on src/app/portfolio/page.tsx:
```diff
@@ -113,15 +112,7 @@ const PortfolioPage = () => {
-                                            Deposit
-                                        </Button>
-                                    </SignInRedirectorLayout>
+                                    <DialogsWrapper variant="default" />
```
why re we calling dailogue wrapper here

Prithvi101 commented on src/app/trade/page.tsx:
```diff
@@ -52,13 +55,7 @@ export default function TradePage() {
-                    >
-                        Deposit
-                    </Button>
+                    <DialogsWrapper variant="trade" />
```
same for here

0xthi committed:
refactor: refactored DialogWrapper and removed it from Trade and Portfolio pages

0xthi committed:
Merge branch 'dev' of https://github.com/Goemon-xyz/goemon-mvp into ui-153-add-conditional-rendering-logic-on-the-app-based-on-network
